### PR TITLE
[sc-9192] AI models section redesign

### PR DESCRIPTION
**Pull request by mpellerin42**
- Fix documentation of nsi-two-columns-configuration-item component
- Fix typography of pages description and title
- New sticky footer component
-  Refactoring of AI models page
    - expanders replaced by tabs
    - no more pdf selection toggle
    - two columns layout for each tab

### CODE REVIEW

File: apps/sistema-demo/src/app/app.component.ts [hunk 1]
```diff
@@ -56,10 +56,11 @@ import {
   SistemaLabelComponent,
   SistemaPasswordInputComponent,
   SistemaSpinnerComponent,
+  SistemaStickyFooterComponent,
+  SistemaTwoColumnsConfigurationItemComponent,
 } from './sistema-pages';
 import { SliderPageComponent } from '../../../../libs/pastanaga-angular/projects/demo/src/app/demo/pages/slider-page/slider-page.component';
```
Comment (ebrehault) [inline apps/sistema-demo/src/app/app.component.ts:62]: wrong import

File: apps/sistema-demo/src/app/sistema-pages/sistema-sticky-footer/sistema-sticky-footer.component.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import { ChangeDetectionStrategy, Component } from '@angular/core';
+import { CommonModule } from '@angular/common';
+import { PaDemoModule } from '../../../../../../libs/pastanaga-angular/projects/demo/src';
```
Comment (ebrehault) [inline apps/sistema-demo/src/app/sistema-pages/sistema-sticky-footer/sistema-sticky-footer.component.ts:3]: wrong import

File: libs/common/src/assets/i18n/fr.json [hunk 3]
```diff
@@ -476,36 +478,43 @@
 	"getting-started.upload.link-textarea.help": "Une URL par ligne",
 	"getting-started.upload.link-textarea.label": "Tapez ou collez les URLs des pages Web",
 	"getting-started.upload.title": "1. Télécharger des données",
-	"kb.ai-models.anonymization_toggle.help": "Appliquer l'anonymisation conforme au Règlement général sur la protection des données (RGPD) à toutes les nouvelles ressources de la Knowledge Box",
-	"kb.ai-models.anonymization_toggle.label": "Anonymisation",
-	"kb.ai-models.confirm-anonymization.confirm-button": "Anonymiser",
-	"kb.ai-models.confirm-anonymization.description": "Une fois les ressources anonymisées, cela ne peut plus être annulé. <br>Remarque : les nouveaux paramètres d'anonymisation n'affecteront pas les ressources existantes",
-	"kb.ai-models.confirm-anonymization.title": "Attention : les ressources futures seront affectées",
+	"kb.ai-models.anonymization.activate.description": "Appliquez l’anonymisation conforme au Règlement général sur la protection des données (RGPD) à toutes les nouvelles ressources de la Knowledge Box.",
```
Comment (ebrehault) [inline libs/common/src/assets/i18n/fr.json:481]: Appliquez => Appliquer
conforme => conformément

File: libs/common/src/assets/i18n/fr.json [hunk 4]
```diff
@@ -476,36 +478,43 @@
...
+	"kb.ai-models.anonymization.confirm-anonymization.title": "Attention : les ressources futures seront affectées",
+	"kb.ai-models.answer-generation.behavior.description": "Définissez les traits de personnalité de l'assistant, comment il doit se comporter lorsqu'il répond à des questions ou lors d'une conversation. Par exemple, vous pourriez définir que le système répondra en tant que professionnel de santé.",
+	"kb.ai-models.answer-generation.behavior.label": "Décrire le comportement",
+	"kb.ai-models.answer-generation.behavior.title": "Personnaliser le comportement",
+	"kb.ai-models.answer-generation.prompt.description": "Définit la manière dont vous souhaitez que le système agisse lorsque vous posez une question. Ici, vous pouvez définir des règles plus spécifiques, qui pourraient même remplacer le comportement défini. Par exemple, vous pouvez définir une invite qui oblige le système à donner uniquement des résultats sous forme de listes, ou même à demander de répondre dans une langue donnée.",
```
Comment (ebrehault) [inline libs/common/src/assets/i18n/fr.json:491]: so far, I think we used "prompt" for all language (so not "invite")

File: libs/common/src/assets/i18n/fr.json [hunk 5]
```diff
@@ -476,36 +478,43 @@
...
+	"kb.ai-models.answer-generation.prompt.title": "Définir une invite",
+	"kb.ai-models.answer-generation.select-llm.description": "Choisissez le LLM qui fonctionnera le mieux pour répondre aux questions de vos utilisateurs.",
+	"kb.ai-models.answer-generation.select-llm.own-key.label": "Utilisez votre propre clé {{provider}}",
+	"kb.ai-models.answer-generation.select-llm.title": "Sélectionner le LLM utilisé pour générer des réponses",
+	"kb.ai-models.common.examples": "Exemples",
+	"kb.ai-models.common.prompt.label": "Votre invite",
```
Comment (ebrehault) [inline libs/common/src/assets/i18n/fr.json:497]: same

File: libs/common/src/assets/i18n/fr.json [hunk 6]
```diff
@@ -476,36 +478,43 @@
...
-	"kb.ai-models.summary.label": "Type de résumé à la demande",
-	"kb.ai-models.summary_model.help": "Indique le modèle utilisé par la fonctionnalité de résumé à la demande",
-	"kb.ai-models.summary_model.label": "Résumé à la demande",
-	"kb.ai-models.summary_prompt.examples": "Exemples",
-	"kb.ai-models.summary_prompt.prompt": "Invite de résumé personnalisée",
-	"kb.ai-models.summary_prompt.prompt_help": "Définit une invite personnalisée qui sera utilisée à la place de l'invite par défaut lors de la génération d'un résumé.",
+	"kb.ai-models.semantic-model.available-models.description": "Le modèle sémantique actuel dépend des langues que vous avez sélectionnées lors de la création de cette Knowledge Box.",
+	"kb.ai-models.semantic-model.available-models.label": "Modèle sémantique",
+	"kb.ai-models.semantic-model.available-models.title": "Modèle sémantique actuel",
+	"kb.ai-models.summarization.prompt.description": "Définit une invite personnalisée qui sera utilisée à la place de l'invite par défaut lors de la génération d'un résumé.",
```
Comment (ebrehault) [inline libs/common/src/assets/i18n/fr.json:506]: same

File: libs/common/src/assets/i18n/fr.json [hunk 7]
```diff
@@ -476,36 +478,43 @@
...
+	"kb.ai-models.summarization.prompt.title": "Définir une invite",
```
Comment (ebrehault) [inline libs/common/src/assets/i18n/fr.json:507]: same

**Review by ebrehault — APPROVED** [review]

File: apps/sistema-demo/src/app/app.component.ts (hunk 1 above)
Comment (mpellerin42) [reply apps/sistema-demo/src/app/app.component.ts:62]: still not, this is how we do the imports of pastanaga document pages in sistema-demo (documentation is in the demo project from pastanaga, which is not exported)

**Review by operramon — APPROVED** [review]

**Review by operramon — APPROVED** [review]

File: libs/common/src/lib/ai-models/answer-generation/answer-generation.component.html [hunk 8]
```diff
@@ -0,0 +1,150 @@
...
+              <pa-toggle
+                labelOnRight
+                [(value)]="hasOwnKey">
+                {{
+                  'kb.ai-models.answer-generation.select-llm.own-key.label'
+                    | translate: { provider: keyProviders[currentGenerativeModelPrompt] }
+                }}
+              </pa-toggle>
+
+              @if (hasOwnKey) {
```
[the remark this answers is not in the document]
Comment (mpellerin42) [reply libs/common/src/lib/ai-models/answer-generation/answer-generation.component.html:39]: weird, it works with chapGPT 3.5 but not with other models 🤔 
I'll investigate deeper

[the remark this answers is not in the document]
Comment (mpellerin42) [reply libs/common/src/lib/ai-models/answer-generation/answer-generation.component.html:39]: it's a good catch @operramon : this bug was actually already there in the current implementation! Basically it's working the first time we save a key, but if you then change the model for one with a user key having the same name as the previous one, then the form is storing the change… I'll work on a fix

**Pull request merged**